src/main/java/com/xjeffrose/chicago/client/RequestMuxer.java

```diff
@@ -45,11 +45,28 @@ public void start() throws Exception {
     blockAndAwaitPool();
     isRunning.set(true);
 
-    workerLoop.scheduleAtFixedRate(() -> {
-      if (messageQ.size() > 0) {
-        drainMessageQ();
+    new Thread(() -> {
+      while (isRunning.get()) {
+        if (messageQ.size() > 0) {
+          messageQ.forEach(xs -> {
+            drainMessageQ();
+            flushPool();
```

Comment:
Should the flushPool call be just after the forEach?  Seems the intent was to prevent flushing on each write but it looks like it will still flush between each write since drainMessage only sends the first message in the queue now.